server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/RequestScopedIdpDefinitionsForZone.java

```diff
@@ -0,0 +1,22 @@
+package org.cloudfoundry.identity.uaa.provider.saml;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.cloudfoundry.identity.uaa.provider.IdentityProvider;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.RequestScope;
+
+@RequestScope
+@Component("requestScopedIdpDefinitionsForZone")
+public class RequestScopedIdpDefinitionsForZone {
```

Comment:
I think it would be better to include "cache" in the name. Consider RequestScopedIdpDefinitionsCache instead

Reply:
Done